models/cice/model_mod.f90

```diff
@@ -431,10 +411,10 @@ subroutine static_init_model()
 ! Determine the shape of the variables from "cice.r.nc"
 ! The assimilate.csh, perfect_model.csh must ensure the cice restart file
 ! is linked to this filename.
-domain_id = add_domain('cice.r.nc', nfields, &
-                       var_names = variable_table(1:nfields, VAR_NAME_INDEX), &
-                       kind_list = state_kinds_list(1:nfields), &
-                       update_list = update_var_list(1:nfields))
+! parse_variables converts the character table that was read in from
+! model_nml:model_state_variables and returns a state_var_type that can be
+! passed to add_domain
+domain_id = add_domain('cice.r.nc', parse_variables(model_state_variables))
```

Comment:
Missing the logic for model_state_variables = ' ' 

https://github.com/NCAR/DART/blob/f2deaa27499fea386d4b72d44a4134f4adf2ab62/models/cice/model_mod.f90#L2618-L2622